src/base/QXmppOmemoData.cpp

```diff
@@ -786,3 +787,90 @@ bool QXmppOmemoElement::isOmemoElement(const QDomElement &element)
     return element.tagName() == QStringLiteral("encrypted") &&
         element.namespaceURI() == ns_omemo_2;
 }
+
+///
+/// \class QXmppOmemoIq
+///
+/// \brief The QXmppOmemoIq class represents an encrypted IQ stanza as defined
+/// by \xep{0384, OMEMO Encryption} and \xep{0420, Stanza Content Encryption}
+/// (SCE).
+///
+/// \since QXmpp 1.5
+///
+/// \ingroup Stanzas
+
+class QXmppOmemoIqPrivate : public QSharedData
+{
+public:
+    QXmppOmemoElement omemoElement;
+};
+
+///
+/// Constructs an OMEMO IQ stanza.
+///
+QXmppOmemoIq::QXmppOmemoIq()
+    : d(new QXmppOmemoIqPrivate)
+{
+}
+
+///
+/// Returns the OMEMO element which contains the data used by OMEMO.
+///
+/// \return the OMEMO element
+///
+QXmppOmemoElement QXmppOmemoIq::omemoElement()
+{
+    return d->omemoElement;
+}
+
+///
+/// Sets the OMEMO element which contains the data used by OMEMO.
+///
+/// \param omemoElement OMEMO element
+///
+void QXmppOmemoIq::setOmemoElement(const QXmppOmemoElement &omemoElement)
+{
+    d->omemoElement = omemoElement;
+}
+
+///
+/// Constructs a copy of \a other.
+///
+/// \param other
+///
+QXmppOmemoIq::QXmppOmemoIq(const QXmppOmemoIq &other) = default;
+
+QXmppOmemoIq::~QXmppOmemoIq() = default;
+
+///
+/// Assigns \a other to this OMEMO IQ stanza.
+///
+/// \param other
+///
+QXmppOmemoIq &QXmppOmemoIq::operator=(const QXmppOmemoIq &other) = default;
+
+/// \cond
+void QXmppOmemoIq::parseElementFromChild(const QDomElement &element)
+{
+    QDomElement child = element.firstChildElement();
+    d->omemoElement.parse(child);
+}
+
+void QXmppOmemoIq::toXmlElementFromChild(QXmlStreamWriter *writer) const
+{
+    d->omemoElement.toXml(writer);
+}
+/// \endcond
+
+///
+/// Determines whether the given DOM element is an OMEMO IQ stanza.
+///
+/// \param element DOM element being checked
+///
+/// \return true if element is an OMEMO IQ stanza, otherwise false
+///
+bool QXmppOmemoIq::isOmemoIq(const QDomElement &element)
+{
+    const QDomElement &child = element.firstChildElement();
```

Comment:
`&` is wrong here, firstChildElement() returns `QDomElement` (no ref)

```suggestion
    auto child = element.firstChildElement();
```